infra/main.tf

```diff
@@ -435,293 +435,102 @@ resource "kubernetes_service" "ukbb_rb_static" {
   }
 }
 
-resource "random_id" "atgu_name_suffix" {
-  byte_length = 2
-}
-
-resource "google_service_account" "atgu" {
-  account_id = "atgu-${random_id.atgu_name_suffix.hex}"
-}
-
-resource "google_service_account_key" "atgu_key" {
-  service_account_id = google_service_account.atgu.name
-}
-
-resource "kubernetes_secret" "atgu_gsa_key" {
-  metadata {
-    name = "atgu-gsa-key"
-  }
-
-  data = {
-    "key.json" = base64decode(google_service_account_key.atgu_key.private_key)
-  }
-}
-
-resource "random_id" "auth_name_suffix" {
-  byte_length = 2
-}
-
-resource "google_service_account" "auth" {
-  account_id = "auth-${random_id.auth_name_suffix.hex}"
-}
-
-resource "google_service_account_key" "auth_key" {
-  service_account_id = google_service_account.auth.name
-}
-
-resource "google_project_iam_member" "auth_service_account_admin" {
-  role = "roles/iam.serviceAccountAdmin"
-  member = "serviceAccount:${google_service_account.auth.email}"
-}
-
-resource "google_project_iam_member" "auth_service_account_key_admin" {
-  role = "roles/iam.serviceAccountKeyAdmin"
-  member = "serviceAccount:${google_service_account.auth.email}"
-}
-
-resource "google_project_iam_member" "auth_storage_admin" {
-  role = "roles/storage.admin"
-  member = "serviceAccount:${google_service_account.auth.email}"
-}
-
-resource "kubernetes_secret" "auth_gsa_key" {
-  metadata {
-    name = "auth-gsa-key"
-  }
-
-  data = {
-    "key.json" = base64decode(google_service_account_key.auth_key.private_key)
-  }
-}
-
-resource "random_id" "batch_name_suffix" {
-  byte_length = 2
-}
-
-resource "google_service_account" "batch" {
-  account_id = "batch-${random_id.batch_name_suffix.hex}"
-}
-
-resource "google_service_account_key" "batch_key" {
-  service_account_id = google_service_account.batch.name
-}
-
-resource "kubernetes_secret" "batch_gsa_key" {
-  metadata {
-    name = "batch-gsa-key"
-  }
-
-  data = {
-    "key.json" = base64decode(google_service_account_key.batch_key.private_key)
-  }
-}
-
-resource "google_project_iam_member" "batch_compute_instance_admin" {
-  role = "roles/compute.instanceAdmin.v1"
-  member = "serviceAccount:${google_service_account.batch.email}"
-}
-
-resource "google_project_iam_member" "batch_service_account_user" {
-  role = "roles/iam.serviceAccountUser"
-  member = "serviceAccount:${google_service_account.batch.email}"
+module "atgu_gsa_secret" {
+  source = "./gsa_k8s_secret"
+  name = "atgu"
 }
 
-resource "google_project_iam_member" "batch_logging_viewer" {
-  role = "roles/logging.viewer"
-  member = "serviceAccount:${google_service_account.batch.email}"
+module "auth_gsa_secret" {
+  source = "./gsa_k8s_secret"
+  name = "auth"
+  iam_roles = [
+    "iam.serviceAccountAdmin",
+    "iam.serviceAccountKeyAdmin",
+    "storage.admin",
```

Comment:
We might be able to get rid of the `storage.admin` role. We shouldn't be creating buckets for users anymore.

Reply:
I'm not to the point where I can run the full test suite in another project and validate this, but I can leave it commented out and remove it in a follow-up if there's no problem? That way if we do use it somewhere I have a clue as to what might be going wrong.